lib/httpDigest.js

```diff
@@ -19,13 +22,54 @@ async function create({data, algorithm = 'sha256', useMultihash = true}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
+  algorithm = algorithm.toLowerCase();
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return _createMultiHash({digest});
+  }
+  return `SHA-256=${base64.encode(digest)}`;
+}
+
+/**
+ * Verifies digest of data created by create().
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.header - The digest header string to
+ *   be verified against.
+ *
+ * @returns {object} - Returns an object with verified true or false.
+ */
+async function verify({data, header}) {
+  if(typeof data !== 'string') {
+    data = JSON.stringify(data);
+  }
   let digest;
+  let encoded;
+  const algorithm = header.split('=')[0].replace('-', '').toLowerCase();
+  if(algorithm === 'sha256') {
+    encoded = header.substring(header.indexOf('=') + 1);
+    digest = await _getDigest({data, algorithm});
+  } else if(algorithm === 'mh') {
+    encoded = header.substring(header.indexOf('=') + 1).slice(1);
+    digest = await _createMultiHash({digest});
+  }
+  const hashedBytes = new Uint8Array(base64url.decode(encoded));
+
+  if(JSON.stringify(digest) !== JSON.stringify(hashedBytes)) {
```

Comment:
Why are we stringifying here instead of comparing bytes? If we're going to compare as strings, we could just encode the digest we produced and compare against the given value, which means we don't have to decode at all, right? Note: I do think a non-timing comparison here is fine because none of the values are secret.

Reply:
I suspect it's because we've had trouble with cross-platform buffer/uint8array comparison without strings. (We ended up using stringify in our crypto libs as well, like https://github.com/digitalbazaar/ed25519-verification-key-2018/blob/master/src/Ed25519VerificationKey2018.js#L315 )

Reply:
@dlongley @dmitrizagidulin yeah I wasn't able figure a way to compare buffer with unit8array, so I converted both to unit8array and compared it.

Reply:
@dmitrizagidulin, please file an issue to fix that verification key thing.

@JSAssassin, let's just encode the digest we get and then compare the encoded strings -- then we can skip decoding.

Reply:
@dlongley Updated - 5647cd2

Reply:
@dlongley 

> @dmitrizagidulin, please file an issue to fix that verification key thing.

Can you say more? What's the part that should be fixed?

Reply:
Filed an issue here: https://github.com/digitalbazaar/ed25519-verification-key-2018/issues/9